lib/util/utils.ts

```diff
@@ -0,0 +1,331 @@
+ * If not provided then the promise chain starts with undefined.
+ *
+ * @return A chain of resolved or rejected promises
+ */
+export function executePromisesSequentially(promiseFactories: Array<any>, kickstart?: any): Promise<any> {
+  let result = Promise.resolve(kickstart);
+  promiseFactories.forEach((promiseFactory) => {
+    result = result.then(promiseFactory);
+  });
+  return result;
+}
+
+/**
+ * Determines whether the given connection string is an iothub connection string.
+ * @param {string} connectionString The connection string.
+ * @return {boolean} boolean.
+ */
+export function isIotHubConnectionString(connectionString: string): boolean {
+  if (!connectionString || typeof connectionString !== "string") {
+    throw new Error("connectionString is a required parameter and must be of type string.");
+  }
+  let result: boolean = false;
+  const model: any = parseConnectionString<any>(connectionString);
+  if (model && model.HostName && model.SharedAccessKey && model.SharedAccessKeyName) {
+    result = true;
+  }
+  return result;
+}
+
+/**
+ * Describes the delivery annotations.
+ * @interface EventHubDeliveryAnnotations
+ */
+export interface EventHubDeliveryAnnotations extends DeliveryAnnotations {
+  /**
+   * @property {string} [last_enqueued_offset] The offset of the last event.
+   */
+  last_enqueued_offset?: string;
+  /**
+   * @property {number} [last_enqueued_sequence_number] The sequence number of the last event.
+   */
+  last_enqueued_sequence_number?: number;
+  /**
+   * @property {number} [last_enqueued_time_utc] The enqueued time of the last event.
+   */
+  last_enqueued_time_utc?: number;
+  /**
+   * @property {number} [runtime_info_retrieval_time_utc] The retrieval time of the last event.
+   */
+  runtime_info_retrieval_time_utc?: number;
+  /**
+   * @property {string} Any unknown delivery annotations.
+   */
+  [x: string]: any;
+}
+
+/**
+ * Describes the delivery annotations.
+ * @interface ServiceBusDeliveryAnnotations
+ */
+export interface ServiceBusDeliveryAnnotations extends DeliveryAnnotations {
+  /**
+   * @property {string} [last_enqueued_offset] The offset of the last event.
+   */
+  last_enqueued_offset?: string;
+  /**
+   * @property {number} [last_enqueued_sequence_number] The sequence number of the last event.
+   */
+  last_enqueued_sequence_number?: number;
+  /**
+   * @property {number} [last_enqueued_time_utc] The enqueued time of the last event.
+   */
+  last_enqueued_time_utc?: number;
+  /**
+   * @property {number} [runtime_info_retrieval_time_utc] The retrieval time of the last event.
+   */
+  runtime_info_retrieval_time_utc?: number;
+  /**
+   * @property {string} Any unknown delivery annotations.
+   */
+  [x: string]: any;
+}
+
+/**
+ * Map containing message attributes that will be held in the message header.
+ * @interface EventHubMessageAnnotations
+ */
+export interface EventHubMessageAnnotations extends MessageAnnotations {
```

Comment:
why not make these types part of the individual client libraries? anything that only applies to event hubs or only applies to service bus or only applies to iot hub is something that forces a dependency inversion where the underlying library has to be updated to reflect implementation details of the parent libraries that consume it as things change

Reply:
That makes sense. I can let this be inside EventHub sdk